precompiles/distribution/tx.go

```diff
@@ -50,6 +50,10 @@ func (p Precompile) ClaimRewards(
 		return nil, err
 	}
 
+	if maxRetrieve > p.stakingKeeper.MaxValidators(ctx) {
+		return nil, fmt.Errorf("maxRetrieve parameter exceeds the number of validators")
```

Comment:
Nit: add the max validators number in error message
```suggestion
		return nil, fmt.Errorf("maxRetrieve parameter exceeds the maximum number of validators (%d)", maxVals)
```